integrationTests/chainSimulator/staking/delegation_test.go

```diff
@@ -750,6 +764,358 @@ func testChainSimulatorCreateNewDelegationContract(t *testing.T, cs chainSimulat
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getTotalActiveStake", nil)
+	require.Nil(t, err)
+	require.Equal(t, expectedTotalStaked, big.NewInt(0).SetBytes(output.ReturnData[0]))
+	require.Equal(t, expectedTopUp, getBLSTopUpValue(t, metachainNode, delegationContractAddress))
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getUserActiveStake", [][]byte{delegatorBBytes})
+	require.Nil(t, err)
+	require.Zero(t, len(output.ReturnData))
+	require.Equal(t, "view function works only for existing delegators", output.ReturnMessage)
+
+	// Step 4: Perform stakeNodes
+
+	txStakeNodes := generateTransaction(validatorOwnerBytes, 2, delegationContractAddress, zeroValue, fmt.Sprintf("stakeNodes@%s", blsKeys[0]), gasLimitForDelegate)
+	stakeNodesTx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(txStakeNodes, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, stakeNodesTx)
+
+	expectedTopUp = expectedTopUp.Sub(expectedTopUp, stakeValue)
+	expectedTopUp = expectedTopUp.Sub(expectedTopUp, stakeValue)
+	require.Zero(t, expectedTopUp.Cmp(getBLSTopUpValue(t, metachainNode, delegationContractAddress)))
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getAllNodeStates", nil)
+	require.Nil(t, err)
+	stakedKeys, notStakedKeys, unStakedKeys := getNodesFromContract(output.ReturnData)
+	require.Equal(t, 1, len(stakedKeys))
+	require.Equal(t, blsKeys[0], hex.EncodeToString(stakedKeys[0]))
+	require.Equal(t, 0, len(notStakedKeys))
+	require.Equal(t, 0, len(unStakedKeys))
+
+	err = cs.GenerateBlocks(2) // allow the metachain to finalize the block that contains the staking of the node
+	require.Nil(t, err)
+
+	testBLSKeyIsInQueueOrAuction(t, metachainNode, delegationContractAddress, blsKeys[0], expectedTopUp)
+
+	tx2delegatorB := generateTransaction(delegatorBBytes, 1, delegationContractAddress, delegateValue, "delegate", gasLimitForDelegate)
+	delegatorBTx2, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(tx2delegatorB, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, delegatorBTx2)
+	assert.Equal(t, delegatorBTx2.SmartContractResults[0].ReturnMessage, "total delegation cap reached")
+
+	// check the tx failed
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getTotalActiveStake", nil)
+	require.Nil(t, err)
+	require.Equal(t, expectedTotalStaked, big.NewInt(0).SetBytes(output.ReturnData[0]))
+	require.Zero(t, expectedTopUp.Cmp(getBLSTopUpValue(t, metachainNode, delegationContractAddress)))
+
+	output, err = executeQuery(cs, core.MetachainShardId, delegationContractAddress, "getUserActiveStake", [][]byte{delegatorBBytes})
+	require.Nil(t, err)
+	require.Zero(t, len(output.ReturnData))
+	require.Equal(t, "view function works only for existing delegators", output.ReturnMessage)
+
+	delegateValue = delegateValue.Sub(delegateValue, oneEGLD) // 500 EGLD
```

Comment:
this is so awkward. Why not set real values here (or consts). (L1025 reads better :) )

Reply:
fixed

Reply:
:+1: